frontends/p4/dontcareArgs.h

```diff
@@ -17,21 +17,21 @@ limitations under the License.
 #ifndef FRONTENDS_P4_DONTCAREARGS_H_
 #define FRONTENDS_P4_DONTCAREARGS_H_
 
+#include "frontends/common/resolveReferences/resolveReferences.h"
 #include "frontends/p4/typeChecking/typeChecker.h"
 #include "ir/ir.h"
 
 namespace P4 {
 
 /// This class replaces don't care arguments (_) with
 /// a temporary which is unused.
-class DontcareArgs : public Transform {
-    ReferenceMap *refMap;
+class DontcareArgs : public Transform, public ResolutionContext {
+    MinimalNameGenerator nameGen;
```

Comment:
Note that you need to apply the MinimalNameGenerator to the IR prior to using it to create new names (so it can record all the existing names).  Otherwise it might create a name that is the same as an existing name.

Reply:
Never mind -- just noticed that you added an init_apply to do just that,

Reply:
Yeah, I'm normally following that pattern. Unless `NameGenerator` is used in downstream passes.